pkg/db/db.go

```diff
@@ -46,9 +45,13 @@ func Open(filename string) (*DB, error) {
 	if err != nil {
 		return nil, err
 	}
-	db.Version, db.Records, db.uncompacted = deserializeDB(bufio.NewReader(f))
-	f.Close()
+	defer f.Close()
+	db.Version, db.Records, db.uncompacted, err = deserializeDB(bufio.NewReader(f))
+	if err != nil {
+		return nil, err
+	}
 	if len(db.Records) == 0 || db.uncompacted/10*9 > len(db.Records) {
+		f.Close()
```

Comment:
If you already did `defer f.Close()`, do you need `f.Close()` again?

Reply:
Added a comment. PTAL.